### PR TITLE
CI simplify check-watermark

### PR DESCRIPTION
**Pull request by MarcoGorelli**
I got a PR in to pre-commit to add a `--negate` flag to `pygrep`, using `check-watermark` as an example use-case :tada:

So...let's use it? It means not having to maintain an extra `scripts/check_watermark.py` file, which can only be a good thing

----

`pygrep` is a Python implementation of grep, so is cross-platform
`--negate` will exit with code 1 if a pattern is not found (in this case, we want the watermark to be found in every notebook)
`--multiline` makes pygrep look through the entire file, rather than just line-by-line

### CODE REVIEW

**Review by AlexAndorra — CHANGES_REQUESTED** [review]
Super nice, thanks @MarcoGorelli !
Just one thing before merging: can you add the minimal pre-commit version requirement to `environment.yml` too please? It's useful for people using conda 😉

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -24,8 +24,10 @@ repos:
     - id: watermark
       name: Check notebooks have watermark
       types: [jupyter]
-      entry: python scripts/check_watermark.py
-      language: python
+      entry: '%load_ext watermark.*%watermark -n -u -v -iv -w'
+      language: pygrep
+      args: [--negate, --multiline]
+      minimum_pre_commit_version: 2.8.0
```
Comment (michaelosthege) [inline .pre-commit-config.yaml:30]: The exception message in `check_watermark.py` had a link to the Wiki page (https://github.com/pymc-devs/pymc3/wiki/PyMC's-Jupyter-Notebook-Style)
Is there a way to report this link here? Or at least append a note like ` (see Wiki)` to the name of the job, so users understand what this check is about.

**Review by michaelosthege — APPROVED** [review]
I have zero experience with `pre-commit`, so take my approval with a grain of salt. Just left one comment :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (MarcoGorelli) [reply .pre-commit-config.yaml:30]: That's a really good point, thanks!

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -24,8 +24,10 @@ repos:
     - id: watermark
       name: Check notebooks have watermark
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml]:
```suggestion
      name: Check notebooks have watermark (see Jupyter style guide from PyMC3 Wiki)
```

**Review by michaelosthege — APPROVED** [review]

**Review by AlexAndorra — APPROVED** [review]
All good now, thanks @MarcoGorelli !

**Pull request merged**